Modules/_struct.c

```diff
@@ -812,11 +812,16 @@ bu_int(_structmodulestate *state, const char *p, const formatdef *f)
     Py_ssize_t i = f->size;
     const unsigned char *bytes = (const unsigned char *)p;
     do {
+        assert(0 <= x);
```

Comment:
These are addressed in https://github.com/python/cpython/pull/96739. Any feedback you have on that PR would be welcome.

Reply:
Thanks, I'll have a look.

The PR description already sounds exactly like the approach I had in mind for fixing these issues.